src/System.Linq.Expressions/tests/DebugViewTests.cs

```diff
@@ -0,0 +1,507 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.Linq.Expressions.Tests
+{
+    public static class DebugViewTests
+    {
+        private static readonly PropertyInfo s_debugView = typeof(Expression).GetProperty("DebugView", BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public);
+
+        [Fact]
+        public static void Constant_Null()
+        {
+            Check("null", Expression.Constant(null));
+            Check("null", Expression.Constant(null, typeof(string)));
+            Check("null", Expression.Constant(null, typeof(int?)));
+       }
+
+        [Fact]
+        public static void Constant_String()
+        {
+            Check("\"bar\"", Expression.Constant("bar", typeof(string)));
+        }
+
+        [Fact]
+        public static void Constant_Primitives_Literals()
+        {
+            Check("False", Expression.Constant(false, typeof(bool)));
+            Check("True", Expression.Constant(true, typeof(bool)));
+            Check("\'a\'", Expression.Constant('a', typeof(char)));
+            Check("42", Expression.Constant(42, typeof(int)));
```

Comment:
Should we do cases like this as theories? e.g. seems this could be:
```C#
[Theory]
[InlineData("False", false, typeof(bool))]
[InlineData("True", true, typeof(bool))]
[InlineData("\'a\'", 'a', typeof(char))]
[InlineData("42", 42, typeof(int))]
public static void Constant_Primitives_Literals(string expected, object input, Type type)
{
    Check(expected, input, type);
}
```
The advantage is greater when there's more code in the body than just a single line, but it at least makes each item its own test so that a failure in one doesn't cause problems for the others.

I'm fine for these cases either way.